caddyhttp/gzip/gzip.go

```diff
@@ -105,6 +105,10 @@ func (w *gzipResponseWriter) WriteHeader(code int) {
 	w.Header().Del("Content-Length")
 	w.Header().Set("Content-Encoding", "gzip")
 	w.Header().Add("Vary", "Accept-Encoding")
+	originalEtag := w.Header().Get("ETag")
+	if originalEtag != "" {
+		w.Header().Set("ETag", "W/"+originalEtag)
```

Comment:
We should not prepend "W/" if it is already a weak ETag; we will have to perform this check here. It should be pretty straightforward to add a test for that case too.